modules/ximgproc/perf/perf_edge_drawing.cpp

```diff
@@ -0,0 +1,93 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html.
+
+#include "perf_precomp.hpp"
+
+namespace opencv_test
+{
+namespace
+{
+/* 1. Define parameter type and test fixture */
+typedef tuple<int, int> EDTestParam;
+typedef TestBaseWithParam<EDTestParam> EdgeDrawing_Detect_Edges;
+typedef TestBaseWithParam<EDTestParam> EdgeDrawing_Detect_Lines;
+typedef TestBaseWithParam<EDTestParam> EdgeDrawing_Detect_Ellipses;
+
+/* 2. Declare the testsuite */
+PERF_TEST_P(EdgeDrawing_Detect_Edges, perf,
+            Combine(Values(0, 1, 2, 3), Values(10, 20, 30, 50, 100, 150, 200, 250)))
```

Comment:
Do we really need to track performance of all these cases? Could we use subset of them here (and below)?

You can create separate full set (e.g, for "development"), but lets keep it disabled by default (see `DISABLED_FULL, OCL_GaussianBlurFixture` from `imgproc`)

Reply:
i will provide performans test later